scipy/stats/tests/test_stats.py

```diff
@@ -7971,3 +7971,63 @@ def test_rename_mode_method(fun, args):
     with pytest.raises(TypeError, match=re.escape(err)):
         fun(*args, method='exact', mode='exact')
 
+
+class TestExpectile:
+    def test_same_as_mean(self):
+        rng = np.random.default_rng(42)
+        x = rng.random(size=20)
+        assert_allclose(stats.expectile(x, alpha=0.5), np.mean(x))
+
+    def test_minimum(self):
+        rng = np.random.default_rng(42)
+        x = rng.random(size=20)
+        assert_allclose(stats.expectile(x, alpha=0), np.amin(x))
+
+    def test_maximum(self):
+        rng = np.random.default_rng(42)
+        x = rng.random(size=20)
+        assert_allclose(stats.expectile(x, alpha=1), np.amax(x))
+
+    @pytest.mark.parametrize("alpha", [0.2, 0.5, 0.8])
+    def test_expectile_properties(self, alpha):
+        """
+        See Section 6 of
+        I. Steinwart, C. Pasin, R.C. Williamson & S. Zhang (2014).
+        "Elicitation and Identification of Properties". COLT.
+        http://proceedings.mlr.press/v35/steinwart14.html
+        """
+        rng = np.random.default_rng(42)
+        n = 20
+        x = rng.normal(size=n)
+
+        # 1. translation equivariant
+        c = rng.exponential()
+        assert_allclose(
+            stats.expectile(x + c, alpha=alpha),
+            stats.expectile(x, alpha=alpha) + c,
+        )
+        assert_allclose(
+            stats.expectile(x - c, alpha=alpha),
+            stats.expectile(x, alpha=alpha) - c,
+        )
+
+        # 2. positively homogeneous
+        assert_allclose(
+            stats.expectile(c * x, alpha=alpha),
+            c * stats.expectile(x, alpha=alpha),
+        )
+
+        # 3. subadditive
+        y = rng.normal(size=n)
+        if alpha >= 0.5:
+            assert (
+                stats.expectile(x + y, alpha=alpha)
+                <= stats.expectile(y, alpha=alpha)
+                + stats.expectile(y, alpha=alpha)
+            )
+        else:
+            assert (
+                stats.expectile(x + y, alpha=alpha)
+                > stats.expectile(y, alpha=alpha)
+                + stats.expectile(y, alpha=alpha)
```

Comment:
Presumably this needs to replace one y to x as above, but in any case the inversion of the comparison here is not clear from the comment `# 3. subadditive`.

Assuming this is intentional, could we do something like:
```
import operator  # at the top

# 3. subadditive
y = rng.normal(size=n)
# comment explaining why we need to switch
op = operator.le if alpha >= 0.5 else operator.gt
assert op(expectile(x+y, ...), expectile(x, ...) + expectile(y, ...))

Reply:
On second thought, if `alpha=0.5` is equivalent to the `mean` case, then this can never be `>`, because the mean is fully additive (i.e. both sub- & superadditive).
```suggestion
                stats.expectile(x + y, alpha=alpha)
                >= stats.expectile(x, alpha=alpha)
                + stats.expectile(y, alpha=alpha)
```

Reply:
> because the mean is fully additive (i.e. both sub- & superadditive).

OK, I managed to confuse myself here with trying to fit this between the `<=` and `>=`, sorry.

If `f == mean`, then for sets `A, B`, `f(A + B) = f({ f(A), f(B) })`. To give an example, if A={1, 3}, B={2, 4},
```
mean(A + B) = mean({1, 2, 3, 4}) = (1 + 2 + 3 + 4) / 4 = 2.5
mean(A) + mean(B) = mean({1, 3}) + mean({2, 4}) = (1+3)/2 + (2+4)/2 = 2 + 3 = 5 ❌
mean({ mean(A), mean(B) }) = mean({2, 3}) = 2.5 ✔️
```

Based on this example, I don't understand how, in general, `expectile(x + y) > expectile(x) + expectile(y)` can ever be true, as the latter will be twice as large (for `alpha` around 0.5 at least). Maybe I'll have to start reading the paper... 😅

Reply:
You're right. I wrote those tests too fast. Those should be tests on the distribution level, i.e. `expectile(X + Y) <= expectile(X) + expectile(Y)` for some rvs `X~F_1` and `Y~F_2`. Fixed in https://github.com/scipy/scipy/pull/17039/commits/6ea8abd39ae7f281fe675b6325273a773a423768.

Reply:
The mean then reads `E[X + Y] = E[X] + E[Y]`.